### PR TITLE
chore: simplify integration tests for v1 endpoints by using admin_secret directly

### PR DESCRIPTION
**Pull request by RogerHYang**
RBAC testing for the endpoints has been consalidated [here](https://github.com/Arize-ai/phoenix/blob/4806ff9d0e6ae0ac374511aa33dfa8c911b72c45/tests/integration/auth/test_auth.py#L1669)

---

> [!NOTE]
> Replace per-user auth in client integration tests with admin_secret, remove role-based permutations, update helpers/assertions (incl. system user ID), and add a document-annotation identifier validation test.
> 
> - **Integration tests (client)**:
>   - Replace user login/API key setup with `_app.admin_secret` across tests (`annotations`, `datasets`, `experiments`, `projects`, `prompts`, `spans`, `users`).
>   - Remove role/user parameterization and RBAC branches; drop imports/fixtures for `_MEMBER`, `_ADMIN`, `_GetUser`, etc.
>   - Update GraphQL helpers to pass `api_key` directly; set expected user to `_SYSTEM_USER_GID` where applicable (e.g., annotations, prompt tags).
>   - Refactor prompt helper `_create_chat_prompt` to accept `api_key` (no user), and adjust tests accordingly.
>   - Simplify projects CRUD test to always allow update/delete using admin context.
>   - Experiments: centralize client/helper setup with admin secret; minor query/pagination helpers added/cleaned.
>   - Datasets and spans: keep behavior the same while switching auth; minor assertion/order/pagination verifications retained.
>   - Users: perform all REST operations via admin client; keep permission and validation checks (e.g., default user protections, email sanitization).
> - **New/adjusted tests**:
>   - Add server-side validation test rejecting non-empty identifiers for document annotations (HTTP 422).
>   - Ensure dataset split and experiment split associations verified via GraphQL using admin auth.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a7df7744d12b839ee67df982083a6e1cc73a828d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: tests/integration/_helpers.py [hunk 1]
```diff
@@ -682,7 +682,7 @@ def handle_request(self, request: httpx.Request) -> httpx.Response:
 
 def _httpx_client(
     app: _AppInfo,
-    auth: Optional[_SecurityArtifact] = None,
+    auth: Optional[Union[_SecurityArtifact, str]] = None,
```
Comment (mikeldking) [inline tests/integration/_helpers.py]: might be worth type aliasing the string?
